src/string.rs

```diff
@@ -545,7 +595,28 @@ impl String {
     }
   }
 
-  // Creates a v8::String from a `&'static [u16]`.
+  /// Creates a `v8::String` from owned bytes.
+  /// The bytes must be Latin-1 or ASCII.
+  /// V8 will take ownership of the buffer and free it when the string is garbage collected.
+  #[inline(always)]
+  pub fn new_external_onebyte<'s>(
+    scope: &mut HandleScope<'s, ()>,
+    buffer: Box<[u8]>,
```

Comment:
can we have a raw api which accepts a custom destructor as well? it would be nice to get these working: https://github.com/denoland/deno/blob/main/cli/napi/js_native_api.rs#L1181-L1234

Reply:
Added